cabal-install/Distribution/Client/CmdRepl.hs

```diff
@@ -13,34 +17,134 @@ module Distribution.Client.CmdRepl (
+         ( SourcePackage(..) )
+import qualified Distribution.Solver.Types.PackageIndex as PackageIndex
+import Distribution.Types.BuildInfo
+         ( BuildInfo(..), emptyBuildInfo )
 import Distribution.Types.ComponentName
-         ( componentNameString )
+         ( ComponentName(..), componentNameString )
+import Distribution.Types.CondTree
+         ( CondTree(..) )
+import Distribution.Types.Dependency
+         ( Dependency(..), thisPackageVersion )
+import Distribution.Types.GenericPackageDescription
+         ( emptyGenericPackageDescription )
+import Distribution.Types.PackageDescription
+         ( PackageDescription(..), emptyPackageDescription )
+import Distribution.Types.Library
+         ( Library(..), emptyLibrary )
+import Distribution.Types.PackageId
+         ( PackageIdentifier(..), PackageId )
+import Distribution.Types.UnqualComponentName
+         ( UnqualComponentName )
+import Distribution.Types.Version
+         ( mkVersion, version0, nullVersion )
+import Distribution.Types.VersionRange
+         ( anyVersion )
 import Distribution.Text
          ( display )
 import Distribution.Verbosity
-         ( Verbosity, normal )
+         ( Verbosity, normal, lessVerbose )
 import Distribution.Simple.Utils
-         ( wrapText, die', ordNub )
-
+         ( wrapText, die', ordNub, createTempDirectory, handleDoesNotExist )
+import Language.Haskell.Extension
+         ( Language(..) )
+
+import Control.Exception
+         ( catch, throwIO )
+import Control.Monad 
+         ( when, unless )
+import Data.List
+         ( sortOn )
 import qualified Data.Map as Map
+import Data.Ord
+        ( Down(..) )
 import qualified Data.Set as Set
-import Control.Monad (when)
-
-
-replCommand :: CommandUI (ConfigFlags, ConfigExFlags, InstallFlags, HaddockFlags, [String])
+import System.Directory
+         ( getTemporaryDirectory, removeDirectoryRecursive )
+import System.FilePath
+         ( (</>) )
+
+type ReplFlags = [String]
+
+data EnvFlags = EnvFlags 
+  { envPackages :: [PackageId]
+  , envIncludeTransitive :: Flag Bool
+  , envOnlySpecified :: Flag Bool
+  }
+
+defaultEnvFlags :: EnvFlags
+defaultEnvFlags = EnvFlags
+  { envPackages = []
+  , envIncludeTransitive = toFlag True
+  , envOnlySpecified = toFlag False
+  }
+
+envOptions :: ShowOrParseArgs -> [OptionField EnvFlags]
+envOptions _ =
+  [ option ['p'] ["package"]
+    "Include an additional package in the environment presented to GHCi."
+    envPackages (\p flags -> flags { envPackages = p ++ envPackages flags })
+    (reqArg "PACKAGE" packageIdReadE (fmap prettyShow :: [PackageId] -> [String]))
+  , option [] ["no-transitive-deps"]
+    "Don't automatically include transitive dependencies of requested packages."
+    envIncludeTransitive (\p flags -> flags { envIncludeTransitive = p })
+    falseArg
+  , option ['z'] ["only-specified"]
+    "Only include explicitly specified packages (and 'base'). This implies '--no-transitive-deps'."
```

Comment:
Description doesn't make it clear how `--only-specified` differs from `--no-transitive-deps`. I guess we should mention environment files somewhere.

Reply:
They differ if you're in a project/package context, `--only-specified` ignores that context.

Reply:
Yep, I figured, but it's not obvious from the description. Maybe we should call it `--ignore-project`?

Reply:
I'm against introducing a new flag for something we can already express with current flags (especially because it makes things awkward when such competing/conflciting flags get turned into cabal.config/cabal.project flags):

We already have `--project-file=`, and I've been arguing that we can just have `--project-file=''` denote to ignore any project context; i.e. force a project-less context.